cuda/private/rules/cuda_library.bzl

```diff
@@ -81,6 +81,18 @@ def _cuda_library_impl(ctx):
     libs = [] if lib == None else [lib]
     pic_libs = [] if pic_lib == None else [pic_lib]
 
+    cc_infos = [
+        CcInfo(
+            compilation_context = compilation_ctx,
+            linking_context = linking_ctx,
+        ),
+    ]
+
+    if hasattr(ctx.attr, "_builtin_deps"):
+        cc_infos.extend([dep[CcInfo] for dep in ctx.attr._builtin_deps if CcInfo in dep])
+
+    cc_info = cc_common.merge_cc_infos(cc_infos = cc_infos, direct_cc_infos = [dep[CcInfo] for dep in ctx.attr.deps if CcInfo in dep])
```

Comment:
I am not sure if this is correct here. The `direct_cc_infos`, as per the docs,

> List of CcInfos to be merged, whose headers will be exported by the direct fields in the returned provider.

I think cc info from deps is not direct.

Reply:
Take https://bazel.build/rules/lib/CompilationContext#direct_headers as an example:
> Returns the list of modular headers that are declared by this target. This includes both public headers (such as those listed in "hdrs") and private headers (such as those listed in "srcs").

Reply:
Yep, agreed; I will revisit this.